### PR TITLE
Add retry logic 

### PR DESCRIPTION
**Pull request by stefan-strandberg**
I've seen that sometimes my tasks seem to stall forever for reasons unknown to me :). To counter this I have added a configurable retry that will reset the current run after a configurable number of seconds.

When working on this I also found an issue where runs would not be loaded after a restart of the service, see comments inline.

### CODE REVIEW

File: scheduler.go [hunk 1]
```diff
@@ -267,7 +305,7 @@ func (s *Scheduler) Init(db *bolt.DB) chan struct{} {
 			return fmt.Errorf("create bucket: %s", err)
 		}
 
-		c := runningBucket.Cursor()
+		c := runsBucket.Cursor()
```
Comment (stefan-strandberg) [inline scheduler.go:311]: Pretty sure this should be runsBucket?

**Comment by stefan-strandberg** [comment]
Found some issues with this PR please hold review for a while :)

**Comment by stefan-strandberg** [comment]
Should be ok for review now

**Comment by jimmybergman** [comment]
1. The part about runningBucket looks fishy, on startup it is supposed to only set the in-memory representation from the persisted version of each running machine, not each machine that has ever run

2. For the retry part to be perfect it would be preferable if we:
a) Had a channel that the scheduler can communicate with the running goroutine while machinePtr.RunningStateCode == true to tell the goroutine to cancel
b) That ExecuteState is changed to not block until execution is done but rather start the process and check for either process completion or the scheduler telling to kill process and the goroutine
c) If the scheduler wants to retry, then instead of just setting RunningStateCode to false, it should communicate this to ExecuteState through the channel and ExecuteState should kill the actual external process, then return from the goroutine and then the retry can be initiated

i.e. so that we don't keep two processes running for the same state concurrently

**Pull request closed without merging**